streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -197,92 +195,254 @@ public boolean isActive() {
+        } else {
+            if (state() == State.RUNNING) {
+                closeTopology(true);
+            }
 
-    private void initializeTaskTime(final Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata) {
-        for (final Map.Entry<TopicPartition, OffsetAndMetadata> entry : offsetsAndMetadata.entrySet()) {
-            final TopicPartition partition = entry.getKey();
-            final OffsetAndMetadata metadata = entry.getValue();
+            if (state() == State.RUNNING || state() == State.RESTORING) {
+                commitState();
+                // whenever we have successfully committed state during suspension, it is safe to checkpoint
+                // the state as well no matter if EOS is enabled or not
+                stateMgr.checkpoint(checkpointableOffsets());
 
-            if (metadata != null) {
-                final long committedTimestamp = decodeTimestamp(metadata.metadata());
-                partitionGroup.setPartitionTime(partition, committedTimestamp);
-                log.debug("A committed timestamp was detected: setting the partition time of partition {}"
-                              + " to {} in stream task {}", partition, committedTimestamp, id);
+                // we should also clear any buffered records of a task when suspending it
+                partitionGroup.clear();
+
+                transitionTo(State.SUSPENDED);
+                log.debug("Suspended running");
             } else {
-                log.debug("No committed timestamp was found in metadata for partition {}", partition);
+                throw new IllegalStateException("Illegal state " + state() + " while suspending active task " + id);
             }
         }
-
-        final Set<TopicPartition> nonCommitted = new HashSet<>(partitions);
-        nonCommitted.removeAll(offsetsAndMetadata.keySet());
-        for (final TopicPartition partition : nonCommitted) {
-            log.debug("No committed offset for partition {}, therefore no timestamp can be found for this partition", partition);
-        }
-    }
-
-    // package private for testing
-    void initializeStateStores() {
-        TaskUtils.registerStateStores(topology, stateDirectory, id, logPrefix, log, processorContext, stateMgr);
     }
 
     /**
      * <pre>
      * - resume the task
      * </pre>
      */
+    @Override
     public void resume() {
         switch (state()) {
+            case CLOSING:
             case RUNNING:
             case RESTORING:
-                // no need to do anything, just let them continue running / restoring
+                // no need to do anything, just let them continue running / restoring / closing
+                log.trace("Skip resuming since state is {}", state());
                 break;
 
             case SUSPENDED:
-                initializeMetadata();
-                transitionTo(State.RUNNING);
+                // just re-initilaize the topology is sufficient
+                initializeTopology();
```

Comment:
This is a discovered bug-fix: in resumption we should re-initialize the topology: previously it is called in AssignedTasks but after the consolidation it was not called anymore.